Sources/Fuzzilli/Core/CodeGenerators.swift

```diff
@@ -255,6 +255,27 @@ public let CodeGenerators: [CodeGenerator] = [
         b.callMethod(methodName!, on: obj, withArgs: arguments)
     },
 
+    CodeGenerator("MethodCallWithSpreadGenerator", input: .object()) { b, obj in
+        guard b.mode != .conservative else { return }
+
+        var methodName = b.type(of: obj).randomMethod()
+        if methodName == nil {
+            guard b.mode != .conservative else { return }
+            methodName = b.genMethodName()
+        }
+
+        var arguments: [Variable] = []
```

Comment:
Thinking about it again, I think we should still try to prefer spreading iterable objects, since attempting to spread most values will lead to exceptions (e.g. `TypeError: Found non-callable @@iterator` for a simple object). So how about turning lines 267-274, and the same code below, into
```swift
var arguments: [Variable] = []
var spreads: [Bool] = []
for _ in 0...Int.random(in: 3...5) {
    let val = b.randVar()
    arguments.append(val)
    // Prefer to spread values that we know are iterable, as non-iterable values will lead to exceptions ("TypeError: Found non-callable @@iterator")
    if b.type(of: val).Is(.iterable) {
        spreads.append(probability(0.9))
    } else {
        spreads.append(probability(0.1))
    }
}
```
And, since this is I think required 4 times now, WDYT about moving it into a new `ProgramBuilder.randCallArgumentsWithSpreading(n: Int) -> (arguments: [Variable], spreads: [Bool])` method?

Reply:
Nice! I like this approach